ordered_set.py

```diff
@@ -112,6 +112,21 @@ def add(self, key):
         return self.map[key]
     append = add
 
+    def update(self, sequence):
+        """
+        Update set with given iterable sequence, then return index of last element.
+        If `sequence` is not iterable then just add this element
+        """
+        try:
+            iter(sequence)
+        except TypeError:
```

Comment:
Why would we need this case? It seems pretty unexpected to me. It doesn't match what `set` does and it seems that it wuold hide bugs.

Passing a non-iterable to `.update()` should be an error.